### ISSUE TITLE
Bogus memory usage reported by EXPLAIN ANALYZE

### ISSUE THREAD

**Issue by soumyadeep2007** [greenplum-db/gpdb#9914]
w/ @d 

### Greenplum version or build
master. **Cannot repro on 6X_STABLE**
### Expected behavior
Executor memory reported for slice1 should be sane.
### Actual behavior
```
                                                       QUERY PLAN
------------------------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)  (cost=3.20..3.68 rows=16 width=8) (actual time=0.487..0.498 rows=16 loops=1)
   ->  HashAggregate  (cost=3.20..3.36 rows=6 width=8) (actual time=0.093..0.140 rows=7 loops=1)
         Group Key: i
         Extra Text: (seg2)   Hash chain length 1.4 avg, 2 max, using 5 of 32 buckets; total 0 expansions.
         ->  Seq Scan on t1  (cost=0.00..3.16 rows=6 width=4) (actual time=0.060..0.062 rows=7 loops=1)
 Planning time: 0.368 ms
   (slice0)    Executor memory: 39K bytes.
   (slice1)    Executor memory: 18014398509481946K bytes avg x 3 workers, 18014398509481946K bytes max (seg0).
 Memory used:  1024kB
 Optimizer: Postgres query optimizer
 Execution time: 0.964 ms
(11 rows)
```
18014398509481946K => garbage
### Step to reproduce the behavior
```sql
create temp table t1(i int);
insert into t1 select generate_series(1, 16);
explain analyze select 1 from t1 group by i;
```

**Comment by hlinnaka** [comment]
Fixed by commit 87c905e95295097a20efaa7d64a8395591c8e50d.

**Comment by d** [comment]
Thanks @hlinnaka !
